lib/AST/ASTVerifier.cpp

```diff
@@ -796,6 +796,13 @@ class Verifier : public ASTWalker {
       if (!shouldVerify(cast<Stmt>(S)))
         return false;
 
+      if (Ctx.LangOpts.hasFeature(Feature::PackIteration)) {
```

Comment:
I don't think we need this many `hasFeature` checks - probably the only place where it needs to show up is CSGen in `generateForEachStmtConstraints`, everything else down the line should just assume that if it got `PackExpansionExpr` there it's valid and enabled (you can make that an assertion if you want).